python/lsst/dax/distribution/DataIngest.py

```diff
@@ -0,0 +1,297 @@
+            return False
+        return True
+
+    def sendTableSchema(self, schema_file_path):
+        with open(schema_file_path, 'r') as schema_f:
+            data = schema_f.read()
+            data_json = json.loads(data)
+        success, r_json = self._postToIngest('ingest/table', data_json)
+        if not success:
+            print('ERROR while sending table schema ', schema_file_path, "r_json=", r_json)
+            return False
+        return True
+
+    def startTransaction(self, db_name):
+        """ Start an ingest super transaction.
+
+        Return
+        ------
+        success : bool
+            True when transaction was started successfully.
+        id : int
+            Ingest super transaction id number.
+        """
+        success, r_json = self._postToIngest('ingest/trans', {'database':db_name,'auth_key':''})
+        if not success:
+            print('ERROR when starting transaction ', db_name, "r_json=", r_json)
+            return False, -1
+        id = r_json['databases'][db_name]['transactions'][0]['id']
+        print("transaction id=", id, " r_json=", r_json)
+        return True, id
+
+    def endTransaction(self, transaction_id, abort):
+        """ End the ingest transaction
+
+        Parameters
+        ----------
+        transaction_id : int
+            The transaction id number.
+        abort : bool
+            True if the transaction should be rolled back.
+
+        Return
+        ------
+        success : bool
+            True if the operation was successful.
+        status : int
+            Status code from the put operation.
+        r_json : json or None
+            json data from the put operation if status was 200.
+        """
+        cmd = 'ingest/trans/' + str(transaction_id) + '?abort='
+        if abort:
+            cmd += '1'
+        else:
+            cmd += '0'
+        success, status, r_json = self._putToIngest(cmd, {'auth_key':self._auth_key})
+        if not success:
+            print("ERROR ending transaction id=", transaction_id, "abort=", abort, "status=", status,
+                  "r_json=", r_json)
+        return success, status, r_json
+
+    def getChunkTargetAddr(self, transaction_id, chunk_id):
+        """ Get the host and port number of the ingest worker for this chunk.
+
+        Parameters
+        ----------
+        transaction_id : int
+            Ingest transaction id number.
+        chunk_id : int
+            Chunk id number.
+
+        Return
+        ------
+        host : str
+            Host name of the ingest worker.
+        port : int
+            Port number of the ingest worker.
+        """
+        cmd = 'ingest/chunk'
+        jdata = {"transaction_id":transaction_id,"chunk":chunk_id,"auth_key":self._auth_key}
+        success, r_json = self._postToIngest(cmd, jdata)
+        if not success:
+            print("ERROR failed to get chunk target address", jdata, " r_json=", r_json)
```

Comment:
This printout duplicates what the exception does. And it bloats your code here. I presume you may just print the message encoded into the exception at a point where you will be catching it upstream.